NETCORE/src/Microsoft.ApplicationInsights.AspNetCore/TelemetryInitializers/SyntheticTelemetryInitializer.cs

```diff
@@ -4,24 +4,66 @@
     using Microsoft.ApplicationInsights.Channel;
     using Microsoft.ApplicationInsights.DataContracts;
     using Microsoft.AspNetCore.Http;
+    using Microsoft.Extensions.Configuration;
 
     /// <summary>
-    /// This will allow to mark synthetic traffic from availability tests.
+    /// This will allow to mark synthetic traffic from availability tests or configured User-Agent filters.
     /// </summary>
     public class SyntheticTelemetryInitializer : TelemetryInitializerBase
     {
         private const string SyntheticTestRunId = "SyntheticTest-RunId";
         private const string SyntheticTestLocation = "SyntheticTest-Location";
-
         private const string SyntheticSourceHeaderValue = "Application Insights Availability Monitoring";
 
+        private const string UserAgentHeader = "User-Agent";
+        private const string SyntheticUserAgentFiltersFromConfig = "ApplicationInsights:SyntheticUserAgentFilters";
+        private const string SyntheticUserAgentSourceName = "Bot";
+        private const string SyntheticUserAgentSourceNameKey = "Microsoft.ApplicationInsights.RequestTelemetry.SyntheticUserAgentSource";
+        private const string SyntheticUserAgentDefaultFilters = "search|spider|crawl|Bot|Monitor|AlwaysOn";
+        private string syntheticUserAgentFilters;
+        private string[] syntheticUserAgentFilterPatterns;
+
         /// <summary>
         /// Initializes a new instance of the <see cref="SyntheticTelemetryInitializer" /> class.
         /// </summary>
         /// <param name="httpContextAccessor">Accessor to provide HttpContext corresponding to telemetry items.</param>
-        public SyntheticTelemetryInitializer(IHttpContextAccessor httpContextAccessor)
+        /// <param name="configuration">Configuration to get synthetic user agent filters.</param>
+        public SyntheticTelemetryInitializer(IHttpContextAccessor httpContextAccessor, IConfiguration configuration)
```

Comment:
This will be breaking change to existing customers. We can add new ctor, but need to keep old one as is., or make new params optional.

Reply:
Hi @cijothomas, I thought about it, indeed, but my understanding is that this would cause issues with dependency injection.
Do you mean existing customers sometimes explicitly call this constructor? (instead of relying on AddAspNetCoreWebTelemetryInitializers)
Thanks

Reply:
> Do you mean existing customers sometimes explicitly call this constructor?

They can. Its a public API. So we have to always maintain the backward compatibility, since we follow semver guidelines.

(could you detail what is the issue DI?)

Reply:
I was wondering how DI would know which constructor to use and looked at https://docs.microsoft.com/en-us/dotnet/core/extensions/dependency-injection#constructor-injection-behavior. 
I also saw https://blogs.cuttingedge.it/steven/posts/2013/di-anti-pattern-multiple-constructors/, but I am not 100% sure if this applies to .NET Core DI.